libsolidity/formal/SMTEncoder.cpp

```diff
@@ -375,30 +375,9 @@ bool SMTEncoder::visit(Assignment const& _assignment)
 {
 	auto const& left = _assignment.leftHandSide();
 	auto const& right = _assignment.rightHandSide();
-
-	if (auto const* memberAccess = isEmptyPush(left))
-	{
-		right.accept(*this);
-		left.accept(*this);
-
-		auto const& memberExpr = memberAccess->expression();
-		auto& symbArray = dynamic_cast<smt::SymbolicArrayVariable&>(*m_context.expression(memberExpr));
-		smtutil::Expression oldElements = symbArray.elements();
-		smtutil::Expression length = symbArray.length();
-		symbArray.increaseIndex();
-		m_context.addAssertion(symbArray.elements() == smtutil::Expression::store(
-			oldElements,
-			length - 1,
-			expr(right)
-		));
-		m_context.addAssertion(symbArray.length() == length);
-
-		arrayPushPopAssign(memberExpr, symbArray.currentValue());
-		defineExpr(_assignment, expr(left));
-		return false;
-	}
-
-	return true;
+	right.accept(*this);
+	left.accept(*this);
+	return false;
```

Comment:
If all this function is doing now is visit left and right and returning false, I think we can remove it entirely, and only rely on `endVisit(Assignment)`

Reply:
I wanted to do that, but that does not work. The point is that we need to first visit RHS and then LHS, but the default is first LHS and then RHS.
I also checked `ExpressionCompiler` and that is also visiting assignment first RHS and then LHS.

Reply:
Ah ok yes, missed that.

Reply:
Added a comment explaining that.